### PR TITLE
Implement the training process using ElasticDL EmbeddingColumn.

### PR DESCRIPTION
**Pull request by brightcoder01**
- Implement the training process of the model with ElasticDL embedding column: Calculate and report the gradients of the embedding values of ElasticDL embedding column.

- Extract a common component `EmbeddingDelegate` to interact with the external EmbeddingStorage such as Parameter Server. It's shared by ElasticDL embedding layer and embedding column.

- Currently we don't update the EmbeddingLayer with the EmbeddingDelegate for the memory leak investigation.

### CODE REVIEW

File: elasticdl/python/worker/worker.py [hunk 1]
```diff
@@ -335,6 +376,17 @@ def report_variable(self):
         for ps_id in range(len(self._ps_stubs)):
             self.report_variable_to_ps(ps_id)
 
+    def collect_edl_embedding_name_values(self):
```
Comment (workingloong) [inline elasticdl/python/worker/worker.py]: Add a docstring to explain the output.

File: elasticdl/python/worker/worker.py [hunk 2]
```diff
@@ -335,6 +376,17 @@ def report_variable(self):
...
+        embedding_name_values = []
+        for layer in self._embedding_layers:
+            embedding_name_values.append((layer.name, layer.embedding_and_ids))
+        for column in self._embedding_columns:
+            embedding_name_values.append(
+                (column.name, column.embedding_and_ids)
+            )
```
Comment (workingloong) [inline elasticdl/python/worker/worker.py:417]: May `layer.name` be the same as `column.name` ?

**Comment by workingloong** [comment]
Add a unit test for ElasticDL embedding_column.

File: elasticdl/python/worker/worker.py (hunk 1 above)
Comment (brightcoder01) [reply elasticdl/python/worker/worker.py]: Good suggestion, added.

File: elasticdl/python/worker/worker.py (hunk 2 above)
Comment (brightcoder01) [reply elasticdl/python/worker/worker.py:417]: The name format for column.name is {column_field}_embedding_elasticdl, the layer name is user defined. I think the probability is low for name conflict.

**Comment by brightcoder01** [comment]
> Add a unit test for ElasticDL embedding_column.

Good suggestion, add more test cases in feature_column_test.py to cover the gradients calculation.

File: elasticdl/python/worker/worker.py (hunk 2 above)
Comment (workingloong) [reply elasticdl/python/worker/worker.py:417]: Whether do we need to check there are name conflicts? If conflicts exist, we can raise an error or warning.

Comment (brightcoder01) [reply elasticdl/python/worker/worker.py:417]: Good suggestion, added the check logic

**Review by workingloong — APPROVED** [review]

**Pull request merged**